src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -205,6 +205,18 @@ private SelectionInfo GetInitialSelectionInfo(SyntaxNode root, SourceText text)
                 };
             }
 
+            var commonRootContainer = commonRoot.FirstAncestorOrSelf<SyntaxNode>(node => node is ClassDeclarationSyntax or RecordDeclarationSyntax or StructDeclarationSyntax);
```

Comment:
Could probably get away with `TypeDeclarationSyntax`, which is the base for all three of these, and `InterfaceDeclarationSyntax`. Its valid to have methods in interfaces too (in C# 8 and above anyway).

Reply:
Agreed on this.

Reply:
(would be good to then add tests for the interface case).